internal/controller/nonadmin_controller.go

```diff
@@ -158,22 +155,19 @@ func ensureRequiredSpecs(deploymentObject *appsv1.Deployment, dpa *oadpv1alpha1.
 		Name:  "WATCH_NAMESPACE",
 		Value: deploymentObject.Namespace,
 	}
-	if len(dpaBackupSpecResourceVersion) == 0 || !reflect.DeepEqual(dpa.Spec.NonAdmin.EnforceBackupSpec, previousEnforcedBackupSpec) {
-		dpaBackupSpecResourceVersion = dpa.GetResourceVersion()
+	if len(dpaResourceVersion) == 0 ||
+		!reflect.DeepEqual(dpa.Spec.NonAdmin.EnforceBackupSpec, previousEnforcedBackupSpec) ||
+		!reflect.DeepEqual(dpa.Spec.NonAdmin.EnforceRestoreSpec, previousEnforcedRestoreSpec) ||
+		!reflect.DeepEqual(dpa.Spec.NonAdmin.GarbageCollectionPeriod, previousGarbageCollectionPeriod) ||
+		!reflect.DeepEqual(dpa.Spec.NonAdmin.BackupSyncPeriod, previousBackupSyncPeriod) {
```

Comment:
```suggestion
		!reflect.DeepEqual(dpa.Spec.NonAdmin, previousDpaNonAdmin) ||
```
Would this also work?

Reply:
I think so, will test

Reply:
updated